### PR TITLE
ci: fix CI reports not uploading

### PR DESCRIPTION
**Pull request by aaronsteers**
This resolves the issue where `airbyte-ci` reports were not being uploaded, and the PR author would have to manually download the job results artifact, then decompress, and then navigate to the report url.

Requires this PR to be merged:

- https://github.com/airbytehq/airbyte/pull/48824

Now, the report URL is printed in the job execution log (as usual for Airbyte CI tests):

> image

For convenient access, it is also posted to the "Summary" page of the GitHub Workflow for easy access, and to prevent having to scan through job logs:

https://github.com/airbytehq/airbyte-python-cdk/actions/runs/12187332389?pr=138

> image

The job artifacts are still downloadable from the bottom of the Summary page as well, in case other test report artifacts are useful - but for most cases, you can simply go to the summary page and click on the report link in the job summary.

> image

## Summary by CodeRabbit

- **New Features**
	- Enhanced reporting capabilities with HTML report URL extraction.
	- Improved clarity in workflow execution outputs.
  
- **Bug Fixes**
	- Clearer output messages when the job is aborted due to no relevant changes.

- **Chores**
	- Updated job permissions to allow creating checks on pull requests.
	- Specified branch for code checkout in testing.
	- Adjusted job name for conditional skipping based on relevant changes.

### CODE REVIEW

File: .github/workflows/connector-tests.yml
Comment (aaronsteers) [inline .github/workflows/connector-tests.yml:124]: TODO: Revert to `master` before merging, after this has merged:

- https://github.com/airbytehq/airbyte/pull/48824

File: .github/workflows/connector-tests.yml
Comment (aaronsteers) [inline .github/workflows/connector-tests.yml:137]: Specifying a bucket name is now sufficient to tell `airbyte-ci` to upload the reports to GCS.

**Review by natikgadzhi — APPROVED** [review]
Can we also make the GH check line link to the html report?

Approving contingent on the comment that you made to revert the thing to master

**Comment by aaronsteers** [comment]
@natikgadzhi - Yes, I think so. I wanted to implement the same but my attempt was reverted here (https://github.com/airbytehq/airbyte-python-cdk/pull/138/commits/a50c1b74f6f4bd9863c8432fa5cfe27210eacbc8) due to a "not found" error when I tested it. I'll create a follow-on issue to try again in another PR...

**Pull request merged**
